tmt/base.py

```diff
@@ -2072,6 +2072,19 @@ def __init__(self, data, name):
         except KeyError:
             self.log = []
 
+        # Handle alternative result interpretation
+        if interpret == "respect":
+            pass
+        elif interpret == "xfail":
+            self.note = f'original_result: {self.result}'
+            self.result = {'fail': 'pass', 'pass': 'fail'}[self.result]
```

Comment:
This will not cover the `error` case (and the rest):

```
  File "/home/psss/git/tmt/tmt/steps/execute/__init__.py", line 272, in check_shell
    return tmt.Result(data, test.name, test.result)
  File "/home/psss/git/tmt/tmt/base.py", line 2080, in __init__
    self.result = {'fail': 'pass', 'pass': 'fail'}[self.result]
KeyError: 'error'
```

I'd say that other results should be kept untouched in the `xfail` scenario.